packages/core/src/components/tooltip/tooltip.tsx

```diff
@@ -114,13 +127,43 @@ export class Tooltip<
         this.popoverRef.current?.reposition();
     }
 
+    protected validateProps(props: TooltipProps<T>) {
+        const childrenCount = React.Children.count(props.children);
+        if (childrenCount > 1) {
+            console.warn(Errors.POPOVER_WARN_TOO_MANY_CHILDREN);
+        }
+        // all other warnings should occur in Popover, not here.
+    }
+
     // any descendant ContextMenus may update this ctxState
     private renderPopover = (ctxState: TooltipContextState) => {
-        const { children, compact, disabled, intent, popoverClassName, ...restProps } = this.props;
+        const { children, content, renderTarget, compact, disabled, intent, popoverClassName, ...restProps } =
+            this.props;
+
         const popoverClasses = classNames(Classes.TOOLTIP, Classes.intentClass(intent), popoverClassName, {
             [Classes.COMPACT]: compact,
         });
 
+        const contentElement = Utils.ensureElement(content);
+        const tooltipId = contentElement?.props?.id ?? React.useMemo(() => Utils.uniqueId("tooltip"), []);
+        const tooltipRole = contentElement?.props?.role ?? "tooltip";
+        const clonedContent =
+            contentElement &&
+            React.cloneElement(contentElement, {
+                role: tooltipRole,
+                id: tooltipId,
```

Comment:
**nit:** The linter is complaining that these keys, are not sorted alphabetically. We can fix by sorting them:
```tsx
React.cloneElement(contentElement, {
    id: tooltipId,
    role: tooltipRole,
});
```